L1Trigger/CSCTriggerPrimitives/interface/CSCPatternBank.h

```diff
@@ -10,26 +10,21 @@
 
 class CSCPatternBank {
 public:
-  typedef std::vector<std::vector<std::vector<int> > > CLCTPatterns;
-
-  typedef int ALCTPatterns[CSCConstants::NUM_ALCT_PATTERNS][CSCConstants::MAX_WIRES_IN_PATTERN];
+  // typedef used for both ALCT and CLCT
+  typedef std::vector<std::vector<std::vector<int> > > LCTPatterns;
```

Comment:
this is better done with std::array

Reply:
Perhaps, in that case I need different typedefs for ALCT and CLCT.

Reply:
A construct like ```typedef std::vector<std::array<std::array<int, CSCConstants::ALCT_PATTERN_WIDTH>, CSCConstants::NUM_LAYERS > > ALCTPatterns;``` might be okay for ALCT, but ```typedef std::vector<std::array<std::array<int, CSCConstants::CLCT_PATTERN_WIDTH + 2>, CSCConstants::NUM_LAYERS > > CLCTPatterns;``` is too wide layers 1-5 for CLCT. ```typedef std::vector<std::array<std::vector<int>, CSCConstants::NUM_LAYERS > > LCTPatterns;``` would work for both without changing the patterns.